### PR TITLE
Within between stats

### PR DESCRIPTION
**Pull request by huddlej**

### CODE REVIEW

File: notebooks/scripts/convert_gb_to_fasta.py
Comment (huddlej) [inline notebooks/scripts/convert_gb_to_fasta.py:1]: @nandsra21 Can we delete this script? I don't see it used anywhere in the workflows, but maybe it is helpful to have around as a utility?

File: seasonal-flu-nextstrain-2018-2020/Snakefile [hunk 1]
```diff
@@ -474,7 +475,7 @@ rule seasonal_flu_test_create_mutation_table:
     input:
         reference = "seasonal-flu-nextstrain-2018-2020/config/reference_h3n2_ha.fasta",
         alignment = "seasonal-flu-nextstrain-2018-2020/results/aligned_sequences.fasta",
-        embedding = "seasonal-flu-nextstrain-2018-2020/results/embed_{method}_sequences.csv",
+        embedding = "seasonal-flu-nextstrain-2018-2020/results/cluster_embed_{method}_sequences.csv",
```
Comment (huddlej) [inline seasonal-flu-nextstrain-2018-2020/Snakefile:478]: Good catch!

File: notebooks/2023-07-24WithinVBetweenFigure.ipynb
Comment (huddlej) [inline notebooks/2023-07-24WithinVBetweenFigure.ipynb:1]: @nandsra21 Can you run this notebook in the workflow with `snakemake --edit docs/figures/total_within_between_stats.png` and clear the contents of all cells?

**Review by huddlej — COMMENTED** [review]
@nandsra21 Thank you for pushing all of these changes! I pushed a few fixes of my own and left some comments for issues I think you're best able to resolve. Let me know if you have questions.

File: notebooks/scripts/convert_gb_to_fasta.py
Comment (nandsra21) [reply notebooks/scripts/convert_gb_to_fasta.py:1]: Pretty sure we used it originally when we were getting sars/mers data from other sources ... should be safe to delete if we want to!

File: notebooks/scripts/within_v_between_stats.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    parser.add_argument("--group-column", required=True, help="the column corresponding to grouping status")
+    parser.add_argument("--output", required=True,  help="path for outputting as a dataframe")
+
+    args = parser.parse_args()
+
+    distance_matrix = pd.read_csv(args.distance_matrix, index_col=0)
+
+    node_data = pd.read_csv(args.metadata, sep="\t" if args.metadata.endswith(".tsv") else ",")
+    node_data.index = node_data["strain"]
+    node_data.drop("strain", axis=1, inplace=True)
```
[the remark this answers is not in the document]
Comment (nandsra21) [reply notebooks/scripts/within_v_between_stats.py:46]: Done

File: seasonal-flu-nextstrain-2018-2020/Snakefile (hunk 1 above)
Comment (nandsra21) [reply seasonal-flu-nextstrain-2018-2020/Snakefile:478]: :D

File: notebooks/scripts/within_v_between_stats.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    node_dict = node_data.transpose().to_dict()
+
+    group_annotations = pd.DataFrame([
+        {"strain": sequence, args.group_column: node_dict[sequence][args.group_column]}
+        for sequence in node_data.index
+    ])
+
+    # ignoring -1 cluster grouping
+    group_annotations = group_annotations[group_annotations[args.group_column] != -1]
```
[the remark this answers is not in the document]
Comment (nandsra21) [reply notebooks/scripts/within_v_between_stats.py:55]: Done

File: sars-cov-2-nextstrain/2022-03-29-final-figures.py.ipynb [hunk 4]
```diff
@@ -434,7 +434,7 @@
    "outputs": [],
    "source": [
     "pca_chart = (pca_charts[0]) & (pca_charts[1])\n",
-    "pca_chart"
+    "pca_chart.configure_axis(grid=False).configure_view(stroke=None)"
```
[the remark this answers is not in the document]
Comment (nandsra21) [reply sars-cov-2-nextstrain/2022-03-29-final-figures.py.ipynb]: This should be implemented now!

File: docs/Snakefile [hunk 5]
```diff
@@ -1,8 +1,50 @@
...
+        "notebooks/2023-07-24WithinVBetweenFigure.ipynb"
+
+rule concat_mutation_tables: 
+    input: 
+        tables=[
+            "seasonal-flu-nextstrain/results/mutation_table.csv",
+            "seasonal-flu-nextstrain-2018-2020/results/mutation_table.csv",
+            "ha-na-nextstrain/results/mutation_table.csv",
+            "sars-cov-2-nextstrain/results/mutation_table_for_Nextstrain_clade.csv",
+            "sars-cov-2-nextstrain/results/mutation_table_for_Nextclade_pango.csv"
```
Comment (huddlej) [inline docs/Snakefile:28]: We need to add late SC2 data here and update disease names to match the input tables below.

File: docs/Snakefile [hunk 6]
```diff
@@ -1,8 +1,50 @@
 rule manuscript:
     input:
         "docs/cartography.pdf",
+        "docs/figures/total_within_between_stats.png",
+        "docs/tables/total_within_between_stats.csv",
+        "docs/figures/within_between_influenza.png"
         #"cartography.html",
```
Comment (huddlej) [inline docs/Snakefile:7]: We should add the within/between for SC2 as a target here, too

**Comment by huddlej** [comment]
I'm going to merge this now and continue refining it on `master`, so we don't have to keep switching between production and development branches.

**Pull request merged**